taf/repository_tool.py

```diff
@@ -173,31 +261,157 @@ def set_metadata_expiration_date(self, role, start_date=datetime.datetime.now(),
     expiration_date = start_date + datetime.timedelta(interval)
     role_obj.expiration = expiration_date
 
-  def write_roles_metadata(self, role, keystore, update_snapshot_and_timestamp=False):
+  def update_snapshot_and_timestmap(self, keystore, **kwargs):
+    """Update snapshot and timestamp metadata.
+
+    Args:
+      - keystore(str): Path to keystore with snapshot and timestamp keys
+      - kwargs(dict): (Optional) Expiration dates and intervals
+
+    Returns:
+      None
+
+    Raises:
+      - InvalidKeyError: If wrong key is used to sign metadata
+      - MetadataUpdateError: If any other error happened during metadata update
     """
-    Generates metadata of just one metadata file, corresponding
-    to the provided role
-    args:
-      repository: a tu repository
-      role: a tuf role
-      keystore: location of the keystore file
-      update_snapshot_and_timestamp: should timestamp and snapshot.json also be updated
+    try:
+      snapshot_date = kwargs.get('snapshot_date', datetime.datetime.now())
```

Comment:
Do you mean having just one `date` and `interval` for both roles? I have added this because we have different default values for each role, so the API should allow passing values for each role. We are still not using those params, so I would keep it like this for now.

Reply:
Because the function is specific and updates snapshot and timestamp (it's not designed to handle any metadata files), I was thinking about something like this:
`update_snapshot_and_timestmap(self, keystore, snapshot_date=datetime.datetime.now(), snapshot_interval=None, timestamp_date=datetime.datetime.now(), timestamp_interval=None)`